src/DataStreams/NullAndDoCopyBlockInputStream.h

```diff
@@ -21,19 +21,10 @@ class NullAndDoCopyBlockInputStream : public IBlockInputStream
 {
 public:
     NullAndDoCopyBlockInputStream(const BlockInputStreamPtr & input_, BlockOutputStreamPtr output_)
+        : input(std::move(input_))
+        , output(std::move(output_))
```

Comment:
Is it true, that `INSERT SELECT` is single threaded? Or maybe now we can construct many `NullAndDoCopyBlockInputStream` for many streams, but not one `NullAndDoCopyBlockInputStream` for many streams?

Reply:
It is single threaded if setting `max_insert_threads > 1`.